src/java-interop/java-interop-util.cc

```diff
@@ -10,6 +10,8 @@ utf16_to_utf8 (const wchar_t *widestr)
 	char *mbstr = static_cast<char*> (calloc (required_size, sizeof (char)));
 	int converted_size = WideCharToMultiByte (CP_UTF8, 0, widestr, -1, mbstr, required_size, NULL, NULL);
 
+	// Hush a compiler warning about unused variable in RELEASE
+	(void)converted_size;
```

Comment:
how does the compiled code look like? wouldn't it be more readable if we #ifdef'ed the variable declaration?

Reply:
I wanted to avoid the #ifdef noise, because you can't just remove the variable conditionally, you'd either have to call `WideCharToMultiByte` twice in the ifdef blocks, or use an ugly "top-down" ifdef of the form:

```cpp
#if defined(DEBUG)
int converted_size =
#endif
WideCharToMultiByte (CP_UTF8, 0, widestr, -1, mbstr, required_size, NULL, NULL);
```

which is an eyesore. Casting the variable to `void` has no effect on generated code - nothing's there since it's a semantic operation with no side effects. The only downside of leaving the variable (and the cause of the warning) is that you waste some local stack space.

Reply:
It is possible to remove whole variable conditionally as the `assert` macro will "hide" its use. This compiles just fine:
```
#include <assert.h>

int main ()
{
#ifndef NDEBUG
  int i;
#endif

  assert(i==0);

  return 0;
}
```

what if we introduce `assert_variable` macro like this?
```#include <assert.h>

#ifdef NDEBUG
#define assert_variable(x) (__ASSERT_VOID_CAST (0))
#else
#define assert_variable(x) x
#endif

int main ()
{
  assert_variable (int i = 1);

  assert (i == 0);

  return 0;
}
```
That would avoid the comments and make nice code?